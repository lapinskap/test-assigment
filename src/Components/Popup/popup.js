import PropTypes from 'prop-types';
import { Modal } from 'reactstrap';
import React from 'react';

const Popup = ({
  id, isOpen, toggle, unmountOnClose, size, className, children,
}) => (
  <Modal isOpen={isOpen} toggle={toggle} unmountOnClose={unmountOnClose} size={size} className={className}>
    <div data-t1={id}>
      {children}
    </div>
  </Modal>
);

Popup.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.string.isRequired,
  toggle: PropTypes.func.isRequired,
  isOpen: PropTypes.bool,
  unmountOnClose: PropTypes.bool,
  size: PropTypes.string,
  className: PropTypes.string,
};

Popup.defaultProps = {
  isOpen: true,
  unmountOnClose: true,
  size: undefined,
  className: undefined,
};

export default Popup;
