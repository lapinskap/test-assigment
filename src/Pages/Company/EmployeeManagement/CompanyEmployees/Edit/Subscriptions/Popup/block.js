import React, { useCallback, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { banksBanksPermissionWrite } from '../../../../../../../utils/RoleBasedSecurity/permissions';
import Form from '../../../../../../../Components/Form';
import __ from '../../../../../../../utils/Translations';
import { dynamicNotification } from '../../../../../../../utils/Notifications';
import { restApiRequest, SUBSCRIPTION_MANAGEMENT_SERVICE } from '../../../../../../../utils/Api';

const Block = ({ subscription, close }) => {
  const [data, setData] = useState({});
  const onChange = useCallback((key, value) => {
    const updatedData = { ...data };
    updatedData[key] = value;
    setData(updatedData);
  }, [data]);

  useEffect(() => {
    setData({
      blockTo: subscription.blockTo,
    });
  }, [subscription]);

  const submit = async () => {
    try {
      const { id } = subscription;
      await restApiRequest(
        SUBSCRIPTION_MANAGEMENT_SERVICE,
        `/employee-subscription-items/${id}/resign`,
        'POST',
        {
          body: data,
        },
        null,
      );
      dynamicNotification(__('Pomyślnie zrezygnowano ze świadczenia.'));
      close(true);
    } catch (e) {
      console.error(e);
      dynamicNotification(e.mesage || __('Nie udało się zrezygnować ze świadczenia.'), 'error');
    }
  };
  return (
    <Form
      id="blockSubscriptionForm"
      data={data}
      config={{
        isInPopup: true,
        togglePopup: () => close(),
        title: 'Zmiana daty blokady świadczenia',
        onSubmit: submit,
        buttons: [
          {
            onClick: () => close(),
            text: 'Zamknij',
            color: 'light',
            id: 'resetOneTimeFormClose',
          },
          {
            text: 'Zmień',
            type: 'submit',
            id: 'resetOneTimeFormSubmit',
            permissions: banksBanksPermissionWrite,
          },
        ],
        defaultOnChange: onChange,
        noCards: true,
        formGroups: [
          {
            formElements: [
              {
                id: 'blockTo',
                label: 'Blokada ponownego wyboru do:',
                type: 'date',
                tooltip: {
                  content: __('Pozostaw puste aby nie blokować ponownego wyboru tego świadczenia.'),
                },
              },
            ],
          },
        ],
      }}
    />
  );
};

Block.propTypes = {
  close: PropTypes.func.isRequired,
  subscription: PropTypes.shape({
    id: PropTypes.string,
    blockTo: PropTypes.string,
  }).isRequired,
};

export default Block;
