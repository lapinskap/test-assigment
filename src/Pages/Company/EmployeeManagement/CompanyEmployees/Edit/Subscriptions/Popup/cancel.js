import React from 'react';
import PropTypes from 'prop-types';
import { Alert } from 'reactstrap';
import { banksBanksPermissionWrite } from '../../../../../../../utils/RoleBasedSecurity/permissions';
import Form from '../../../../../../../Components/Form';
import __ from '../../../../../../../utils/Translations';
import { dynamicNotification } from '../../../../../../../utils/Notifications';
import { restApiRequest, SUBSCRIPTION_MANAGEMENT_SERVICE } from '../../../../../../../utils/Api';

const Cancel = ({ subscription, close }) => {
  const submit = async () => {
    try {
      const { id } = subscription;
      await restApiRequest(
        SUBSCRIPTION_MANAGEMENT_SERVICE,
        `/employee-subscription-items/${id}/cancel`,
        'POST',
        {
          returnNull: true,
        },
        null,
      );
      dynamicNotification(__('Pomyślnie anulowano świadczenie.'));
      close(true);
    } catch (e) {
      console.error(e);
      dynamicNotification(e.mesage || __('Nie udało się anulować świadczenia.'), 'error');
    }
  };
  return (
    <Form
      id="cancelSubscriptionForm"
      data={{}}
      config={{
        isInPopup: true,
        togglePopup: () => close(),
        title: 'Anulacja świadczenia abonamentowego',
        onSubmit: submit,
        buttons: [
          {
            onClick: () => close(),
            text: 'Zamknij',
            color: 'light',
            id: 'resetOneTimeFormClose',
          },
          {
            text: 'Anulacja świadczenia abonamentowego',
            type: 'submit',
            id: 'resetOneTimeFormSubmit',
            permissions: banksBanksPermissionWrite,
          },
        ],
        defaultOnChange: () => {},
        noCards: true,
        formGroups: [
          {
            formElements: [
              {
                component: Info,
              },
            ],
          },
        ],
      }}
    />
  );
};

Cancel.propTypes = {
  close: PropTypes.func.isRequired,
  subscription: PropTypes.shape({
    id: PropTypes.string,
  }).isRequired,
};

const Info = (
  <Alert key="cancelInfo" color="primary">
    <h4>{__('Uwaga: Anulacja świadczenia jest nieodwracalna.')}</h4>
    <p>
      {__('Konsekwencje anulowania świadczenia')}
      :
    </p>
    <ul>
      <li>{__('data startu świadczenia równa dacie końca świadczenia P = R')}</li>
      <li>
        {__(
          // eslint-disable-next-line max-len
          'kwota w świadczeniu: Przekazanie punktów do banku punktów nie jest korygowana automatycznie. Operator musi zmienić tę kwotę sam',
        )}
      </li>
      <li>
        {__(
          // eslint-disable-next-line max-len
          'liczba punktów w banku punktów nie jest korygowana. Po anulacji świadczenia operator musi poprawić liczbę punktów w banku (np. jeżeli świadczenie anulowane trwało rok, to musi policzyć ile tych punktów powinien dodać',
        )}
      </li>
    </ul>
  </Alert>
);

export default Cancel;
