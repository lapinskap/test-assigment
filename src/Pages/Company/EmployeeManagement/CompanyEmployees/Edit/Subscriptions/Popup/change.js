import React, { useCallback, useEffect, useState } from 'react';
import { Alert } from 'reactstrap';
import PropTypes from 'prop-types';
import { banksBanksPermissionWrite } from '../../../../../../../utils/RoleBasedSecurity/permissions';
import Form from '../../../../../../../Components/Form';
import __ from '../../../../../../../utils/Translations';
import { dynamicNotification } from '../../../../../../../utils/Notifications';
import { restApiRequest, SUBSCRIPTION_MANAGEMENT_SERVICE } from '../../../../../../../utils/Api';

const Change = ({ subscription, close }) => {
  const [data, setData] = useState({});
  const onChange = useCallback((key, value) => {
    const updatedData = { ...data };
    updatedData[key] = value;
    setData(updatedData);
  }, [data]);

  useEffect(() => {
    setData({
      startsAt: subscription.startsAt,
      endsAt: subscription.endsAt,
    });
  }, [subscription]);

  const submit = async () => {
    try {
      const { id } = subscription;
      await restApiRequest(
        SUBSCRIPTION_MANAGEMENT_SERVICE,
        `/employee-subscription-items/${id}/change-dates`,
        'POST',
        {
          body: data,
        },
        null,
      );
      dynamicNotification(__('Pomyślnie zmieniono świadczenie.'));
      close(true);
    } catch (e) {
      console.error(e);
      dynamicNotification(e.mesage || __('Nie udało się zmienić świadczenia.'), 'error');
    }
  };
  return (
    <Form
      id="changeSubscriptionForm"
      data={data}
      config={{
        isInPopup: true,
        togglePopup: () => close(),
        title: 'Zmiana daty obowiązywania świadczenia',
        onSubmit: submit,
        buttons: [
          {
            onClick: () => close(),
            text: 'Zamknij',
            color: 'light',
            id: 'resetOneTimeFormClose',
          },
          {
            text: 'Zmień',
            type: 'submit',
            id: 'resetOneTimeFormSubmit',
            permissions: banksBanksPermissionWrite,
          },
        ],
        defaultOnChange: onChange,
        noCards: true,
        formGroups: [
          {
            formElements: [
              {
                component: Info,
              },
              {
                id: 'startsAt',
                label: 'Nowa data startu:',
                type: 'month',
                validation: ['required'],
              },
              {
                id: 'endsAt',
                label: 'Nowa data końca:',
                type: 'month',
                validation: [{
                  method: 'greaterEqualThanDate',
                  args: [data?.startsAt || 0],
                }],
              },
            ],
          },
        ],
      }}
    />
  );
};

Change.propTypes = {
  close: PropTypes.func.isRequired,
  subscription: PropTypes.shape({
    id: PropTypes.string,
    startsAt: PropTypes.string,
    endsAt: PropTypes.string,
  }).isRequired,
};

const Info = (
  <Alert key="cancelInfo" color="primary">
    <h4>{__('UWAGA: Anulacja świadczenia spowoduje zmiany w raporcie płacowym.')}</h4>
    <p>
      {__(
        // eslint-disable-next-line max-len
        'Czy akceptujesz zmiany na swoją odpowiedzialność oraz czy poinformowałeś/aś wszystkie osoby korzystające z raportu płacowego o zmianach (np. AHR firmy)?',
      )}
    </p>
    <ul>
      <li>{__('pusta data końca oznacza świadczenie na czas nieokreślony')}</li>
      <li>
        {__('świadczenia zależne nie przestawiają się automatycznie')}
      </li>
    </ul>
    <p>
      {__('Zmiana dat na świadczeniu Przekazanie Punktów do Banku Punktów powoduje zmianę saldo na koncie użytkownika:')}
    </p>
    <ul>
      <li>{__('wydłużenie świadczenia doładowuje punkty (sprawdź czy doładowana liczba punktów jest prawidłowa)')}</li>
      <li>
        {__(
          // eslint-disable-next-line max-len
          'skrócenie świadczenia odejmuje punkty (sprawdź czy odjęta liczba punktów jest prawidłowa, jeżeli na koncie użytkownika nie ma wystarczającej liczby punktów potrzebnej do skorygowania daty świadczenia, zmiana daty nie jest możliwa)',
        )}
      </li>
    </ul>
  </Alert>
);

export default Change;
