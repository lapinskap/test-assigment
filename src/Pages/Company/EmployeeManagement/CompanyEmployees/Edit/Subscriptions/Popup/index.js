import React from 'react';
import PropTypes from 'prop-types';
import PopupComponent from '../../../../../../../Components/Popup/popup';
import {
  POPUP_TYPE_BLOCK,
  POPUP_TYPE_CANCEL, POPUP_TYPE_CHANGE, POPUP_TYPE_RESIGN, POPUP_TYPE_SUSPEND,
} from '../utils';
import Cancel from './cancel';
import Suspend from './suspend';
import Resign from './resign';
import Change from './change';
import Block from './block';

export default function Popup({
  close, subscription, type,
}) {
  let content = null;
  if (type === POPUP_TYPE_SUSPEND) {
    content = <Suspend subscription={subscription} close={close} />;
  } else if (type === POPUP_TYPE_CHANGE) {
    content = <Change subscription={subscription} close={close} />;
  } else if (type === POPUP_TYPE_RESIGN) {
    content = <Resign subscription={subscription} close={close} />;
  } else if (type === POPUP_TYPE_CANCEL) {
    content = <Cancel subscription={subscription} close={close} />;
  } else if (type === POPUP_TYPE_BLOCK) {
    content = <Block subscription={subscription} close={close} />;
  }

  if (!content) {
    return null;
  }

  return (
    <PopupComponent id={`${type}Popup`} isOpen toggle={() => close()} unmountOnClose size="lg">
      {content}
    </PopupComponent>
  );
}

Popup.propTypes = {
  close: PropTypes.func.isRequired,
  type: PropTypes.string.isRequired,
  subscription: PropTypes.shape({
    name: PropTypes.string,
    type: PropTypes.string,
    id: PropTypes.string,
  }).isRequired,
};
