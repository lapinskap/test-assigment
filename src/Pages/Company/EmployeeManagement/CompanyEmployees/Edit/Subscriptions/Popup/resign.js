import React, { useCallback, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { banksBanksPermissionWrite } from '../../../../../../../utils/RoleBasedSecurity/permissions';
import Form from '../../../../../../../Components/Form';
import __ from '../../../../../../../utils/Translations';
import { dynamicNotification } from '../../../../../../../utils/Notifications';
import { restApiRequest, SUBSCRIPTION_MANAGEMENT_SERVICE } from '../../../../../../../utils/Api';

const Resign = ({ subscription, close }) => {
  const [data, setData] = useState({});
  const onChange = useCallback((key, value) => {
    const updatedData = { ...data };
    updatedData[key] = value;
    setData(updatedData);
  }, [data]);

  useEffect(() => {
    setData({
      blockTo: subscription.blockTo,
      endsAt: new Date(),
    });
  }, [subscription]);

  const submit = async () => {
    try {
      const { id } = subscription;
      await restApiRequest(
        SUBSCRIPTION_MANAGEMENT_SERVICE,
        `/employee-subscription-items/${id}/resign`,
        'POST',
        {
          body: data,
        },
        null,
      );
      dynamicNotification(__('Pomyślnie zrezygnowano ze świadczenia.'));
      close(true);
    } catch (e) {
      console.error(e);
      dynamicNotification(e.mesage || __('Nie udało się zrezygnować ze świadczenia.'), 'error');
    }
  };
  return (
    <Form
      id="suspendSubscriptionForm"
      data={data}
      config={{
        isInPopup: true,
        togglePopup: () => close(),
        title: 'Zmiana daty zawieszenia świadczenia',
        onSubmit: submit,
        buttons: [
          {
            onClick: () => close(),
            text: 'Zamknij',
            color: 'light',
            id: 'resetOneTimeFormClose',
          },
          {
            text: 'Zmień',
            type: 'submit',
            id: 'resetOneTimeFormSubmit',
            permissions: banksBanksPermissionWrite,
          },
        ],
        defaultOnChange: onChange,
        noCards: true,
        formGroups: [
          {
            formElements: [
              {
                id: 'endsAt',
                label: 'Rezygnacja od:',
                type: 'month',
              },
              {
                id: 'blockTo',
                label: 'Blokada ponownego wyboru do:',
                type: 'date',
                tooltip: {
                  content: __('Pozostaw puste aby nie blokować ponownego wyboru tego świadczenia.'),
                },
                validation: [{
                  method: 'greaterEqualThanDate',
                  args: [data?.endsAt || 0],
                }],
              },
            ],
          },
        ],
      }}
    />
  );
};

Resign.propTypes = {
  close: PropTypes.func.isRequired,
  subscription: PropTypes.shape({
    id: PropTypes.string,
    blockTo: PropTypes.string,
  }).isRequired,
};

export default Resign;
