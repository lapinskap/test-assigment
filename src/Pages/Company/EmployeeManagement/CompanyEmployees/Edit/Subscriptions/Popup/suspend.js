import React, { useCallback, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { banksBanksPermissionWrite } from '../../../../../../../utils/RoleBasedSecurity/permissions';
import Form from '../../../../../../../Components/Form';
import __ from '../../../../../../../utils/Translations';
import { dynamicNotification } from '../../../../../../../utils/Notifications';
import { restApiRequest, SUBSCRIPTION_MANAGEMENT_SERVICE } from '../../../../../../../utils/Api';

const Suspend = ({ subscription, close }) => {
  const [data, setData] = useState({});
  const onChange = useCallback((key, value) => {
    const updatedData = { ...data };
    updatedData[key] = value;
    setData(updatedData);
  }, [data]);

  useEffect(() => {
    setData({
      suspendedFrom: subscription.suspendedFrom,
      suspendedTo: subscription.suspendedTo,
    });
  }, [subscription]);

  const submit = async () => {
    try {
      const { id } = subscription;
      await restApiRequest(
        SUBSCRIPTION_MANAGEMENT_SERVICE,
        `/employee-subscription-items/${id}`,
        'PATCH',
        {
          body: data,
        },
        null,
      );
      dynamicNotification(__('Pomyślnie zmieniono daty zawieszeń świadczenia.'));
      close(true);
    } catch (e) {
      console.error(e);
      dynamicNotification(e.mesage || __('Nie udało się zmienić świadczenia.'), 'error');
    }
  };
  return (
    <Form
      id="suspendSubscriptionForm"
      data={data}
      config={{
        isInPopup: true,
        togglePopup: () => close(),
        title: 'Zmiana daty zawieszenia świadczenia',
        onSubmit: submit,
        buttons: [
          {
            onClick: () => close(),
            text: 'Zamknij',
            color: 'light',
            id: 'resetOneTimeFormClose',
          },
          {
            text: 'Zmień',
            type: 'submit',
            id: 'resetOneTimeFormSubmit',
            permissions: banksBanksPermissionWrite,
          },
        ],
        defaultOnChange: onChange,
        noCards: true,
        formGroups: [
          {
            formElements: [
              {
                id: 'suspendedFrom',
                label: 'Zawieszenie od:',
                type: 'date',
              },
              {
                id: 'suspendedTo',
                label: 'Zawieszenie do:',
                type: 'date',
                validation: [{
                  method: 'greaterEqualThanDate',
                  args: [data?.suspendedFrom || 0],
                }],
              },
            ],
          },
        ],
      }}
    />
  );
};

Suspend.propTypes = {
  close: PropTypes.func.isRequired,
  subscription: PropTypes.shape({
    id: PropTypes.string,
    suspendedFrom: PropTypes.string,
    suspendedTo: PropTypes.string,
  }).isRequired,
};

export default Suspend;
