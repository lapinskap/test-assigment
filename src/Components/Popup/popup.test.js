import React from 'react';
import { mount } from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';
import Popup from './popup';
import '../../tests/setupTests';

describe('<Popup />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should render correct in default state', () => {
    const subject = mount(<Popup id="testPopup" toggle={() => {}} />);
    expect(EnzymeToJson(subject)).toMatchSnapshot();
  });
});
