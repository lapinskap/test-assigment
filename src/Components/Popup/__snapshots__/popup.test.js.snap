// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popup /> Should render correct in default state 1`] = `
<Popup
  id="testPopup"
  isOpen={true}
  toggle={[Function]}
  unmountOnClose={true}
>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    container="body"
    fade={true}
    isOpen={true}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    returnFocusAfterClose={true}
    role="dialog"
    scrollable={false}
    toggle={[Function]}
    unmountOnClose={true}
    zIndex={1050}
  >
    <Portal
      node={
        <div
          style="position: relative; z-index: 1050; display: block;"
          tabindex="-1"
        >
          <div
            class=""
          >
            <div
              class="modal fade"
              role="dialog"
              style="display: block;"
              tabindex="-1"
            >
              <div
                class="modal-dialog"
                role="document"
              >
                <div
                  class="modal-content"
                >
                  <div
                    data-t1="testPopup"
                  />
                </div>
              </div>
            </div>
            <div
              class="modal-backdrop fade"
            />
          </div>
        </div>
      }
    >
      <Portal
        containerInfo={
          <div
            style="position: relative; z-index: 1050; display: block;"
            tabindex="-1"
          >
            <div
              class=""
            >
              <div
                class="modal fade"
                role="dialog"
                style="display: block;"
                tabindex="-1"
              >
                <div
                  class="modal-dialog"
                  role="document"
                >
                  <div
                    class="modal-content"
                  >
                    <div
                      data-t1="testPopup"
                    />
                  </div>
                </div>
              </div>
              <div
                class="modal-backdrop fade"
              />
            </div>
          </div>
        }
      >
        <div
          className=""
        >
          <Fade
            appear={true}
            baseClass="fade"
            baseClassActive="show"
            className="modal"
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={false}
            onClick={[Function]}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            role="dialog"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex="-1"
            tag="div"
            timeout={300}
            unmountOnExit={false}
          >
            <Transition
              appear={true}
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={300}
              unmountOnExit={false}
            >
              <div
                className="modal fade"
                onClick={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                role="dialog"
                style={
                  Object {
                    "display": "block",
                  }
                }
                tabIndex="-1"
              >
                <div
                  className="modal-dialog"
                  role="document"
                >
                  <div
                    className="modal-content"
                  >
                    <div
                      data-t1="testPopup"
                    />
                  </div>
                </div>
              </div>
            </Transition>
          </Fade>
          <Fade
            appear={true}
            baseClass="fade"
            baseClassActive="show"
            className="modal-backdrop"
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={150}
            unmountOnExit={false}
          >
            <Transition
              appear={true}
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={150}
              unmountOnExit={false}
            >
              <div
                className="modal-backdrop fade"
              />
            </Transition>
          </Fade>
        </div>
      </Portal>
    </Portal>
  </Modal>
</Popup>
`;
